### ISSUE TITLE
Rule collision when using svelte bindings

### ISSUE THREAD

**Issue by knd775** [sveltejs/eslint-plugin-svelte#460]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

8.39.0

### What version of `eslint-plugin-svelte` are you using?

2.27.2

### What did you do?

<details>
<summary>Configuration</summary>

```
module.exports = {
  root: true,
  parser: '@typescript-eslint/parser',
  extends: ['eslint:recommended', 'plugin:@typescript-eslint/recommended', 'plugin:svelte/recommended', 'prettier'],
  plugins: ['@typescript-eslint'],
  ignorePatterns: ['*.cjs'],
  parserOptions: {
    sourceType: 'module',
    ecmaVersion: 2020,
    extraFileExtensions: ['.svelte']
  },
  overrides: [
    {
      files: ['*.svelte'],
      parser: 'svelte-eslint-parser',
      parserOptions: {
        parser: '@typescript-eslint/parser'
      }
    }
  ],
  env: {
    browser: true,
    es2017: true,
    node: true
  },
  rules: {
    '@typescript-eslint/no-inferrable-types': 'off',
    '@typescript-eslint/no-explicit-any': 'off',
    '@typescript-eslint/no-unused-vars': [
      'warn',
      {
        argsIgnorePattern: '^_',
        varsIgnorePattern: '^_',
        caughtErrorsIgnorePattern: '^_'
      }
    ],
    '@typescript-eslint/naming-convention': [
      'warn',
      {
        selector: 'default',
        format: ['camelCase', 'snake_case'],
        leadingUnderscore: 'allow',
        trailingUnderscore: 'allow'
      },
      {
        selector: 'variable',
        format: ['camelCase', 'UPPER_CASE'],
        leadingUnderscore: 'allow',
        trailingUnderscore: 'allow'
      },
      {
        selector: 'variable',
        modifiers: ['exported'],
        format: ['camelCase', 'PascalCase']
      },
      {
        selector: 'typeLike',
        format: ['PascalCase']
      },
      {
        selector: 'enumMember',
        format: null
      },
      {
        selector: ['objectLiteralProperty', 'objectLiteralMethod'],
        format: null
      }
    ],
    'no-console': ['warn', { allow: ['warn', 'error'] }],
    'eol-last': 'warn',
    'no-trailing-spaces': 'warn',
    'no-tabs': 'error',
    'prefer-template': 'warn',
    'no-var': 'warn',
    'prefer-const': 'warn',
    'no-const-assign': 'error',
    'no-irregular-whitespace': 'warn',
    quotes: ['warn', 'single', { avoidEscape: true }],
    curly: ['warn', 'multi-line'],
    'no-debugger': 'warn',
    'func-style': ['warn', 'expression', { allowArrowFunctions: true }],
    'svelte/no-at-html-tags': 'off'
  }
};
```
</details>

```svelte
<script>
  let page = {
    offset: 0,
    limit: 10,
    size: 0,
    amounts: [10, 25]
  };
</script>

<MyComponent bind:settings={page} />
```

### What did you expect to happen?

The eslint `prefer-const` rule should be overridden by this plugin when it is being used by a svelte binding.

### What actually happened?

When using `let`:
> 'page' is never reassigned. Use 'const' instead. `eslint(prefer-const)`

When using `const`:
> Cannot bind to a variable which is not writable `svelte(invalid-binding)`

### Link to **GitHub Repo** with Minimal Reproducible Example

I've tried and failed to create a minimal reproduction.

### Additional comments

May be related to https://github.com/sveltejs/eslint-plugin-svelte/issues/404

**Comment by ota-meshi** [comment]
I can't reproduce your problem. Please provide a repository where we can reproduce it.
